docs/getting_started/async-execution-mode.rst

```diff
@@ -0,0 +1,200 @@
+.. _async-execution-mode:
+
+.. title:: Getting Started with Deferrable Operator
+
+Airflow Async Execution Mode
+============================
+
+The Airflow async execution mode in Cosmos designed to improve pipeline performance. This execution mode could be preferred when you’ve long running resources and you want to run them asynchronously by leveraging Airflow’s `deferrable operators <https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/deferring.html>`__. In this mode, additional operators—``SetupAsyncOperator`` and ``TeardownAsyncOperator``—are added to your workflow.
+
+- **SetupAsyncOperator:** This task runs a mocked ``dbt run`` command on your dbt project, which outputs compiled SQL files to the project’s target directory. These compiled SQLs are then uploaded to a remote location specified by the ``remote_target_path`` configuration.
+- **TeardownAsyncOperator:** This task deletes the resources created by ``SetupAsyncOperator`` from the remote location defined by the ``remote_target_path`` configuration.
+
+Advantages of Airflow Async Mode
+++++++++++++++++++++++++++++++++
+
+- **Improved Task Throughput:** Async tasks free up Airflow workers by leveraging the Airflow Trigger framework. While long-running SQL transformations are executing in the data warehouse, the worker is released and can handle other tasks, increasing overall task throughput.
+- **Faster Task Execution:** With Cosmos ``SetupAsyncOperator``, transformations can be executed without invoking the full dbt run command. This reduces the overhead of running unnecessary dbt commands, improving the performance of each task.
```

Comment:
It would be nice to add here on in the above block that the model tasks download and use the previously pre-compile SQL made available by the setup task.